index.rb

```diff
@@ -0,0 +1,120 @@
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    list = is_a?(Range) ? to_a : self
+    i = 0
+    while i < list.length
+      yield(list[i])
+      i += 1
+    end
+    list
+  end
+
+  def my_each_with_index()
+    return enum_for unless block_given?
+
+    list = is_a?(Range) ? to_a : self
+    index = 0
+    while index < list.length
+      yield(self[index], index)
+      index += 1
+    end
+  end
+
+  def my_select
+    return enum_for unless block_given?
+
+    filter = []
+    my_each { |element| filter.push(element) if yield(element) }
+    filter
+  end
+
+  def my_all?(*args)
+    lever = true
+    if !args[0].nil?
+      my_each { |element| lever = false unless args[0] == element }
+    elsif !block_given?
+      my_each { |element| lever = false unless element }
+    else
+      my_each { |element| lever = false unless yield(element) }
+    end
+    lever
+  end
+
+  def my_any?(*arg)
+    lever = false
+    if !arg[0].nil?
+      my_each { |element| lever = true if arg[0] == element }
+    elsif !block_given?
+      my_each { |element| lever = true if element }
+    else
+      my_each { |element| lever = true if yield(element) }
+    end
+    lever
+  end
```

Comment:
- [ ] should have working `my_any?` identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

 #### my_any? 

- [ ] when a class is passed as an argument returns true if at least one of the collection is a member of such class

```ruby
[1, 2.5, 'a', 9].my_any?(Integer) == [1, 2.5, 'a', 9].any?(Integer) => true
```
- Yours is retrieving the wrong answer
![image](https://user-images.githubusercontent.com/52209067/79290462-677bb000-7e89-11ea-8cc5-fc13776f3949.png)

- Real any?
![image](https://user-images.githubusercontent.com/52209067/77689595-8f04ea00-6f67-11ea-9f51-4536c3f8049b.png)

- [ ] when a Regex is passed as an argument returns false if none of the collection matches the Regex

```ruby
%w[dog door rod blade].my_any?(/o/) == %w[dog door rod blade].any?(/o/) => true
```
- Yours retrieves a NoMethodError
![image](https://user-images.githubusercontent.com/52209067/79290179-a2311880-7e88-11ea-9233-f451010362f7.png)

- Real any?
![image](https://user-images.githubusercontent.com/52209067/79290193-aeb57100-7e88-11ea-814e-f5f1014ebfa6.png)

Reply:
solved